core/src/main/java/org/jdbi/v3/core/statement/Batch.java

```diff
@@ -31,15 +29,23 @@
  */
 public class Batch extends BaseStatement<Batch>
 {
+    public static Batch create(Handle handle) {
+        ConfigRegistry config = handle.getConfig().createCopy();
+        return new Batch(config,
+                handle.getConnection(),
+                handle.getStatementBuilder(),
+                new StatementContext(config, handle.getExtensionMethod()));
```

Comment:
Batch, Call, Query, and Config have really similar implementations for this factory method.

I wonder if we could push all this repetitive code into the BaseStatement constructor.